Sources/Stats/mc_stats.cc

```diff
@@ -106,6 +107,7 @@ Stats Statistics(Eigen::Ref<const Eigen::VectorXcd> values,
   NETKET_CHECK(values.size() >= local_number_chains, InvalidInputError,
                "not enough samples to compute statistics");
   constexpr auto NaN = std::numeric_limits<double>::quiet_NaN();
+  constexpr auto iNaN = std::numeric_limits<int>::quiet_NaN();
```

Comment:
Even though this might be legal (i.e. compiles) there is no such thing as an integer NaN. See e.g. https://stackoverflow.com/a/3949459 if you don't trust me :) Also `iNaN == iNaN` will return `true` which should not be the case for NaN.